src/main/java/catan/services/util/play/EndTurnUtil.java

```diff
@@ -16,9 +48,24 @@ public Integer getNextMoveInPreparationStage(GameBean game) {
         }
     }
 
-    public Integer getNextMoveInMainStage(GameBean game) {
-        return game.getCurrentMove().equals(game.getGameUsers().size())
-                ? 1
-                : game.getCurrentMove() + 1;
+    private boolean isCycleFinished(GameBean game) {
+        Integer preparationCycle = game.getPreparationCycle();
+        return isFirstPlayer(game) && !isOddCycle(preparationCycle) || isLastPlayer(game) && isOddCycle(preparationCycle);
+    }
+
+    private boolean isLastCycle(GameBean game, Integer initialBuildingsSet) {
```

Comment:
actually this method does not give you an answer on question if cycle is last, it just compares 2 values that are coming to method, if I pass another value rather then size of cycles, it will return true, but it doesn;t mean that this is last cycle. We should remove this method and use variable instead with name 'isLastCycle'